support/Python/DirectoryStructure.py

```diff
@@ -126,3 +126,99 @@ def list_segments(segments_dir: Union[str, Path]) -> List[Segment]:
         return []
     matches = map(Segment.match, segments_dir.iterdir())
     return sorted(match for match in matches if match)
+
+
+@dataclass(frozen=True, order=True)
+class PipelineStep:
+    """A complete simulation iteration
```

Comment:
How about this:

> A step in a pipeline
>
> Each pipeline step is a numbered directory with a label, e.g. "000_InitialData".
> It can contain a simulation or a pre- or post-processing step such as archiving.
> Here is an example for the directory structure: